frontend/resourceadm/components/ImportResourceModal/ServiceContent/ResourceContent/ResourceContent.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { ReactNode, useState } from 'react';
+import classes from './ResourceContent.module.css';
+import { Altinn2LinkService } from 'app-shared/types/Altinn2LinkService';
+import { useTranslation } from 'react-i18next';
+import { ResourceNameAndId } from 'resourceadm/components/ResourceNameAndId';
+
+export type ResourceContentProps = {
+  altinn2LinkService: Altinn2LinkService;
+  resourceIdExists: boolean;
+};
+
+/**
+ * @component
+ *    Displays the Resource content in the import resource from Altinn 2 modal
+ *
+ * @property {Altinn2LinkService}[altinn2LinkService] - The service to import from
+ * @property {boolean}[resourceIdExists] - If the id already exists
+ *
+ * @returns {ReactNode} - The rendered component
+ */
+export const ResourceContent = ({
+  altinn2LinkService,
+  resourceIdExists,
+}: ResourceContentProps): ReactNode => {
+  const { t } = useTranslation();
+
+  const [id, setId] = useState(altinn2LinkService.serviceName);
+  const [title, setTitle] = useState(altinn2LinkService.serviceName);
+
+  const [editIdFieldOpen, setEditIdFieldOpen] = useState(false);
+  const [bothFieldsHaveSameValue, setBothFieldsHaveSameValue] = useState(true);
+
+  /**
+   * Replaces the spaces in the value typed with '-'.
+   */
+  const handleIDInput = (val: string) => {
+    setId(val.replace(/\s/g, '-'));
+  };
+
+  /**
+   * Updates the value of the title. If the edit field is not open,
+   * then it updates the ID to the same as the title.
+   *
+   * @param val the title value typed
+   */
+  const handleEditTitle = (val: string) => {
+    if (!editIdFieldOpen && bothFieldsHaveSameValue) {
+      setId(val.replace(/\s/g, '-'));
+    }
+    setTitle(val);
+  };
+
+  /**
+   * Handles the click of the edit button. If we click the edit button
+   * so that it closes the edit field, the id is set to the title.
+   *
+   * @param isOpened the value of the button when it is pressed
+   */
+  const handleClickEditButton = (isOpened: boolean, isSave: boolean) => {
+    setEditIdFieldOpen(isOpened);
+
+    if (isSave) {
+      setBothFieldsHaveSameValue(false);
+    } else {
+      if (!isOpened) {
+        setBothFieldsHaveSameValue(true);
+        // If we stop editing, set the ID to the title
+        if (title !== id) setId(title.replace(/\s/g, '-'));
+      }
+    }
+  };
```

Comment:
Could we use the "block-pattern" to avoid nested statements more than necessary? 

```suggestion
  const handleClickEditButton = (isOpened: boolean, isSave: boolean) => {
    setEditIdFieldOpen(isOpened);

    if (isSave) {
      setBothFieldsHaveSameValue(false);
      return;
    }

   if (!isOpened) {
     setBothFieldsHaveSameValue(true);

     const shouldSetTitleToId = title !== id;
     if (shouldSetTitleToId ) {
       setId(title.replace(/\s/g, '-'));
     }
  }
};
```